### PR TITLE
Easier access to GitHub Packages 

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:

### Improves getting artifacts from GitHub Packages Maven repositories 

It is no longer needed to specify repositories by their names. The new syntax for adding repositories standard to Spine SDK for `buildscript` is:

```kotlin
buildscript {
    standardSpineSdkRepositories()
    ...
}
```
And for project repositories:

```kotlin
repositories.standardToSpineSdk()
```
### Consolidates Kotlin extension functions required for builds
... under `buildSrc/.../BuildExtensions.kt` file.  Previously functions were separated in small groups in several files.

Having all the Kotlin extensions functions for build in one file makes it 1) easier to find them 2) sync. with `tests/buildSrc`.

### CODE REVIEW

File: buildSrc/src/main/kotlin/io/spine/internal/gradle/RunGradle.kt [hunk 1]
```diff
@@ -47,6 +47,15 @@ import org.gradle.internal.os.OperatingSystem
...
 
+    companion object {
+
+        /**
+         * Default Gradle build timeout.
+         */
+        private const val BUILD_TIMEOUT_MINUTES: Long = 10
+    }
+
+
```
Comment (armiol) [inline buildSrc/src/main/kotlin/io/spine/internal/gradle/RunGradle.kt:58]: Please remove this redundant line.

File: buildSrc/src/main/kotlin/io/spine/internal/gradle/publish/IncrementGuard.kt [hunk 2]
```diff
@@ -73,20 +73,36 @@ class IncrementGuard : Plugin<Project> {
      * to a feature branch.
      *
      * Returns `false` if the associated reference is not a branch (e.g. a tag) or if it has
-     * the name which ends with `master`. So, on branches such as `master` and `2.x-jdk8-master`
-     * this method would return `false`.
+     * the name which ends with `master` or `main`.
+     *
+     * For example, on the following branhces the method would return `false`:
```
Comment (armiol) [inline buildSrc/src/main/kotlin/io/spine/internal/gradle/publish/IncrementGuard.kt:78]: "branches"

File: settings.gradle.kts [hunk 3]
```diff
@@ -33,7 +33,7 @@ include(
     "test-env",
     "codegen-java",
     "gradle-api",
-    "gradle-plugin"
+    "gradle-plugin",
```
Comment (armiol) [inline settings.gradle.kts:36]: I would kill this extra comma. Just to keep things tidy.

**Review by armiol — APPROVED** [review]
@alexander-yevsyukov 
LGTM with some minor comments to address. 

In general, all the recent PRs like this one are really "I cannot see trees behind this forest" things. There just so many "routine" changes from `config`, so that the significant changes aren't visible. 

Maybe we could split "routine" changes to gazillion `buildSrc` files from those updates really interesting in terms of newly introduced improvements — into separate PRs?

**Comment by alexander-yevsyukov** [comment]
> I cannot see trees behind this forest

As discussed and agreed via GChat: let's apply `config` changes in separate PRs, and then introduce changes under `buildSrc`.

@yevhenii-nadtochii, FYI. Also, please apply changes from this PR to `config`... :)

**Pull request merged**